### PR TITLE
[Core] Implement simple ShopperContext

### PR DESCRIPTION
**Pull request by pjedrzejewski**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Related tickets | -
| License         | MIT

All services, form types, etc. should be stateless, but sometimes we need to have access (in routing, or other contexts) to currently active channel, currency, locale and customer. This service simplifies this and I have also added this info to the web debug toolbar. So as a developer, you can always check the currently selected context.

This is the first step to clean up all the contexts together and unify their behavior and usage.

zrzut ekranu 2016-06-07 o 23 20 22

### CODE REVIEW

File: src/Sylius/Component/Core/Context/ShopperContextInterface.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+     */
+    public function getCurrencyCode();
+
+    /**
+     * @return string
+     */
+    public function getLocaleCode();
+
+    /**
+     * @return CustomerInterface
```
Comment (michalmarcinkowski) [inline src/Sylius/Component/Core/Context/ShopperContextInterface.php]: `CustomerInterface|null`

File: src/Sylius/Component/Core/Context/ShopperContextInterface.php [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Core\Model\CustomerInterface;
+
+/**
+ * @author Paweł Jędrzejewski <pawel@sylius.org>
+ */
+interface ShopperContextInterface
+{
+    /**
+     * @return ChannelInterface
```
Comment (michalmarcinkowski) [inline src/Sylius/Component/Core/Context/ShopperContextInterface.php]: Missing `@throws ...`

File: src/Sylius/Component/Core/Context/ShopperContext.php [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+     */
+    private $customerContext;
+
+    /**
+     * @param ChannelContextInterface $channelContext
+     * @param CurrencyContextInterface $currencyContext
+     * @param LocaleContextInterface $localeContext
+     * @param CustomerContextInterface $customerContext
+     */
+    public function __construct(ChannelContextInterface $channelContext, CurrencyContextInterface $currencyContext, LocaleContextInterface $localeContext, CustomerContextInterface $customerContext)
```
Comment (michalmarcinkowski) [inline src/Sylius/Component/Core/Context/ShopperContext.php]: Too long line.

File: src/Sylius/Component/Core/Context/ShopperContextInterface.php [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+namespace Sylius\Component\Core\Context;
+
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Core\Model\CustomerInterface;
+
+/**
+ * @author Paweł Jędrzejewski <pawel@sylius.org>
+ */
+interface ShopperContextInterface
```
Comment (pamil) [inline src/Sylius/Component/Core/Context/ShopperContextInterface.php]: What do you think about normalizing methods in current interfaces (one has `getCurrentX()`, the other one - `getX()`) and then just extending them here?

Comment (pjedrzejewski) [inline src/Sylius/Component/Core/Context/ShopperContextInterface.php]: @pamil I think it could be good, but let's leave that for a separate PR, we definitely need to normalize methods on all these contexts.

**Comment by michalmarcinkowski** [comment]
Nice! 👍
